### ISSUE TITLE
Pause and resume scan for big drives

### ISSUE THREAD

**Issue by vto80** [Lazza/RecuperaBit#34]
A way to pause and resume scanning of big drives would be a cool feature. 
I've started RecuperaBit on a USB3 8TB drive (that i've overwritten the first 1-2GB) and it's taking a while. So far it's about 2/3 done but it's still going to be another day or two until it finishes. Would not be a problem usually but i've stupidly forgot to put it in a screen and now i'm stuck waiting for it to finish :/

**Comment by NicolasCARPi** [comment]
I don't think this should be part of the software, as you can already pause and resume processes (use Ctrl-z). In my use case, it's 4 Tb drives and yes it takes a long time, and what I think would be better to have is a progress bar of some sort, I believe this should be doable because we know from the start the size of the drive and we could update the progress bar after each sector scan.

> I've stupidly forgot to put it in a screen and now i'm stuck

Note that you can attach a process started outside screen in screen (or tmux for modern people :p). See [reptyr](https://github.com/nelhage/reptyr) ;)

**Comment by beaster99** [comment]
I'm not sure that is the full ask from vto80.
The Save file does not seem to be written out to disk until the scan is completed.
I've had numerous cases of the scan handing indefinitely after ~1.2TB of a 2 TB disk recovery. Each time that happens, there is no method to resume from a break-point that is manually invoked because of some need. There needs to be some value in the save file being able to reduce the rescanning work as is the case with TestDisk and some of the commercial tools.

**Comment by Lazza** [comment]
Both the suggestion of a progress bar _and_ better handling of progress are good features. I agree that they should be included in some form.

Please note that **currently** the textual interface is the least polished part of the software. A lot of work was put into the file system reconstruction inner workings and not so much on the initial scanning / TUI.

However, improvements can surely be made!

>  The Save file does not seem to be written out to disk until the scan is completed

It should be written if you interrupt with Ctrl+C. But in any case, I agree that it's not ideal.

What I am thinking about is that the scan file should be abandoned in favor of a more complete status/project file in SQLite form.

The first step (except finding the time 😛) would be to draft the structure of such a file.

This would be also related to #63 since all the UI parts can be simply discarded and rewritten. Only the file system module (which is large) should be migrated to Python 3.
